### PR TITLE
Project Restructure and initial UWP App

### PR DESCRIPTION
**Pull request by chuckries**
This commit restructures the project slightly yet again.

1. Move all build files out of src. The top level windows dir contains
all build and source files specific to windows platforms. Nes is now built
as a static library. The motivation for this was so that both the dll
project and the native windows runtime component could simply link against
the same static lib rather than try to include all the same files in their
projects.

2. Initial implementation of Nes as a native windows runtime component.
This provides a simple C++/CX wrapper that allows Nes to be constructed
and used inside of any windows runtime application.
TODO: Improve the asynchronicity of the API.

3. Initial UWP for Nes. The initial UWP is written in C# and references
the native windows runtime component. Today it opens a file picker
immediately when the app starts and will start playing the nes file that
is chosen. It also adds file association for .nes and enough simple code
to make file activation work.
TODO: Lots. This is a pretty barebones app that is the base to continue
building a full featured emulator application from.

4. Add configurations for x86, x64, and ARM. For phone scenarios we need
an ARM build and for UWP I imagine we will want primarily x64. I am fine
if we only support debug support on x86 if that is easier, but I wanted to
be able to at least build everything.

### CODE REVIEW

**Comment by chuckries** [comment]
Sorry that this is all in one giant commit but I realized I needed the static lib after the windows runtime component was already in place and it made the most sense to just bring everything in together.

**Comment by plnelson** [comment]
Why do we want x64?

File: include/nes_api.h [hunk 1]
```diff
@@ -1,11 +1,11 @@
 #pragma once
 
-#include "baseinterface.h"
-#include "nptr.h"
+#include "nes_interfaces.h"
 #include "object.h"
+#include "nptr.h"
 
 #if defined(_WIN32)
```
Comment (plnelson) [inline include/nes_api.h]: Why are we using dllexport from a static lib?

**Comment by chuckries** [comment]
For completeness. I also thought I heard something said about xbox requiring 64 bit when UWP support is released, but I'm not sure.

File: include/nes_api.h (hunk 1 above)
Comment (chuckries) [inline include/nes_api.h]: nes_api.h is not included in the static lib.

File: src/stdafx.h [hunk 2]
```diff
@@ -25,7 +25,7 @@
 #include <experimental/filesystem>
 namespace fs = std::experimental::filesystem;
 
-#include "../include/baseinterface.h"
+#include "../include/nes_interfaces.h"
```
Comment (chuckries) [inline src/stdafx.h]: makes sense, I'll change it.

**Comment by plnelson** [comment]
As an aside, the DAC build will also have to include all of the source files again.  Unfortunately the lib solution will not work for that.

**Comment by chuckries** [comment]
I wonder if it would be worth breaking out the ClCompile and ClInclude ItemGroups into it's own proj file and having both the static lib and the DAC projects reference it?

**Comment by chuckries** [comment]
Although in that world adding new items from the IDE becomes a problem.

**Comment by plnelson** [comment]
I've also thought about changing the project files to include *.cpp, but that also means adding files through the IDE does the wrong thing

**Comment by chuckries** [comment]
Yeah, I'd prefer to avoid that if possible.

**Pull request merged**